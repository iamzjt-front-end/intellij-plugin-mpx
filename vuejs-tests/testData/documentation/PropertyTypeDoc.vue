<template>
  <span :class="{'add-me-if-true': foo, 'add-if-bar': bar}">
    Foo
  </span>
  <FooBar :f<caret>oo="12" :bar="23">

  </FooBar>
</template>
<script lang="ts">
import { defineComponent } from 'vue';

const useSomeProps = (foo) => ({
  foo: {
    type: [String, Array, Object],
    required: true,
  },
});

export default defineComponent({
                                 name:"FooBar",
                                 props: {
                                   ...useSomeProps('foobar'),
                                   bar: {
                                     type: Boolean,
                                     default: true,
                                   },
                                 },
                               });
</script>