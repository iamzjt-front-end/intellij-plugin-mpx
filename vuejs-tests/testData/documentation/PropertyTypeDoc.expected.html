<div class='definition'><pre>foo:&#32;<span style="color:#660e7a;"></span><span style=""></span><span style="color:#000080;">string&#32;</span><span style="">|&#32;</span><span style="color:#000080;">any</span><span style="">[]&#32;|&#32;Object</span></pre></div>
<table class='sections'>
<tr><td valign='top' class='section'><p>Required</td>
</table>